### ISSUE TITLE
adding -delete option to remove the cert files

### ISSUE THREAD

**Issue by quenenni** [certbot/certbot#2551]
Hello,

Do you think you can add an -delete option to let's encrypt?

My context is this: 
From a hosting control panel, users will be able to create an owncloud instance.
When deleting their owncloud instance, I should delete also the certificates files in /etc/letsencrypt folder (I'm not talking about revocation as I can let the certs expire).

Deleting them seems important as by not deleting myself the files in /etc/letsencrypt/{archives,live,renewal}, if I understand correctly, the renewal of the certs will renew all the certs, included the certs for (sub)domains that don't exist anymore.

I must say I'm not sure how the renewal system works, if it tests the domain validity or the host file presence before renewing it. But still, you can have a host file for a domain name that doesn't exist anymore (not very good, but it can happen) and you can have a catchall for subdomains that show a generic page or use the (sub)domain for another purpose.

The only post I found on that topic (https://community.letsencrypt.org/t/correct-way-to-completely-remove-issued-certificate-s-for-a-domain/7409/4) doesn't have an answer on this problem.

If I manage the deletion by myself, deleting all the files /etc/letsencrypt/ subdirs, what about if something changes in the way let's encrypt manage the files (names / placements / ..).
The script will be broken.

Maybe there is another way:
If, by using the revoke option, the renewal feature won't renew the revoked certs, I can live with that.
But in that case, if later, a user want to re-use the same (sub)domain, is it possible to reactivate the certs? (didn't see anything about that in the doc. The renew feature doesn't say anything about this case)

So, is it possible to add such a feature?

And by the way, let's encrypt is a wonderful project. Thanks to you all.

**Comment by pde** [comment]
Cc: @schoen 

Yes, we should definitely have this. It's probably best to implement after we finish the cert manager ui that jdkasten [started](https://github.com/letsencrypt/letsencrypt/tree/cert_manager) a long time ago but never quite landed.

As for workarounds in the meantime...

You're right that in the long run things may change on disk, causing scripts to break.  If something does change in the filesystem layout, it's much more likely to be rearrangement of the structure below `/etc/letsencrypt/archive`, than it is to be below `/etc/letsencrypt/live/` or `/etc/letsencrypt/renewal`.  And perhaps we can ensure that if `archive` does change semantics, the new version will have a new directory name too.

**Comment by quenenni** [comment]
Thanks for your quick response pde.

I'm in no hurry.
In the meantime, I'll manage the deletions of the files / dirs myself and will be happy to change my scripts when the option is added.

**Comment by schoen** [comment]
> Thanks for your quick response pde.
> 
> I'm in no hurry.
> In the meantime, I'll manage the deletions of the files / dirs myself and will be happy to change my scripts when the option is added.

If you're sure that a particular cert will never be used again (for
example, you're planning to delete the only copy of the private key),
you can revoke it first with letsencrypt revoke. That reduces the risk
that someone else could somehow reconstruct the private key later (like
from a mislaid backup or by compromising hosting infrastructure).

After the cert has been revoked, it should be safe to delete the
associated files on your server as long as you delete all of the files
that relate to a certificate lineage at roughly the same time (that
is, the renewal configuration file and the particular lineage-related
directories under live and archive).

The cert manager should be able to help with all of this once we get
it updated and released.

**Comment by quenenni** [comment]
Thanks for your help schoen.

Well.. Nothing is never so sure..
You know.. Some people will delete their onwcloud instance, sure they won't want / need it anymore, but the next month, they will want it again.

In that case, what's the best way to do it?
- Revoke the cert, then delete the cert files, then re-create the cert the next month (with a new private key, that's not a problem, isn't it? Just the problem with the quota)?
- Just delete the cert files and re-create the cert the next month (new private key and problem with the quota)?
- Just delete the Apache host file and let the cert files on the server (with the problem with auto-renewal system that will try to renew unused certs, but no problem with the quota)?
- Just remove the conf file in /etc/letsencrypt/renewal/ in order to not have the auto renewal process to renew that domain (is it working that way?) and remove the Apache host file. And the next month, move back the renew conf file and renew the cert?

I try to understand what the revocation process implies, what if I want to 'un-revoke' a cert?
Is it possible?
-> https://letsencrypt.github.io/acme-spec/, but they only explain what happen when you revoke a cert, not when you want to re-use that (sub)domain and so, re-use / recreate / renew a cert that was previously revoked.
In short, is the revocation process linked to the key only? Or to the domain? Or both?

**Comment by nitmir** [comment]
The revocation in only tided up to a certificate. It is not possible to un-revoke a certificate. If you want to reuse the domain for https, you just have to request a new certificate for the domain.

**Comment by quenenni** [comment]
Excellent. Thanks nitmir.

So I guess my first point should be good.
I'm going to do few tests.

**Comment by bmw** [comment]
See #3615.

**Comment by dchmelik** [comment]
I'd like to see that usable for 'lineages,' which I don't use, and since am not quite good using certbot, end up with more than other people... for any site, I only want /etc/letsencrypt/live/example.com/cert.pem , not a symlink to ../../archive/certn.pem for various n.
